pymodbus/transport/transport.py

```diff
@@ -176,39 +176,37 @@ def __init__(
         else:
             host = self.comm_params.host
             port = int(self.comm_params.port)
-        if self.comm_params.comm_type == CommType.SERIAL:
-            host, port = self.init_setup_serial(host, port)
-            if not host and not port:
-                return
-        if host == NULLMODEM_HOST:
+        if NULLMODEM_HOST in host:
+            if self.comm_params.comm_type == CommType.SERIAL:
+                host, port = NULLMODEM_HOST, int(host[9:].split(":")[1])
             self.call_create = lambda: self.create_nullmodem(port)
             return
-        # TCP/TLS/UDP
-        self.init_setup_connect_listen(host, port)
-
-    def init_setup_serial(self, host: str, _port: int) -> tuple[str, int]:
-        """Split host for serial if needed."""
-        if NULLMODEM_HOST in host:
-            return NULLMODEM_HOST, int(host[9:].split(":")[1])
-        if self.is_server and host.startswith("socket"):
+        if (
+            self.comm_params.comm_type == CommType.SERIAL
+            and self.is_server
+            and host.startswith("socket")
+        ):
             # format is "socket://<host>:port"
             self.comm_params.comm_type = CommType.TCP
+            host = host.strip("socket://")
```

Comment:
the old code was more compact, and did the job, so why change it ?

Reply:
clarity

Reply:
well that is not more clear to me...now I have to make sure nothing references host after it is changed.

Reply:
Would you prefer below?
```python
parts = host.strip("socket://").split(":")
```

Reply:
That is at least still a one liner, but I fail to understand what is wrong with the original code, that only indexed the string.

I am not in favor of making changes just for the sake of changing... months from now when bisecting you will wonder if there was a reason why it was changed and if the old code was faulty.